packages/lit-dev-content/site/tutorials/content/async-directive/05.md

```diff
@@ -0,0 +1,77 @@
+To make our directive update asynchronously, we'll need to start a timer that
+re-renders the time elapsed.
+
+Add a function that starts a periodic 1s timer if it's not already started.
+Since directives are stateful, we can use a class field to store the timer's
+handle, and use that to start the timer only once:
+
+{% switchable-sample %}
+
+```ts
+  timer: number | undefined;
+
+  ensureTimerStarted() {
+    if (this.timer === undefined) {
+      this.timer = setInterval(() => {
+        /* do some periodic work */
+      }, 1000);
+    }
+  }
+```
+
+```js
+  timer = undefined;
+
+  ensureTimerStarted() {
+    if (this.timer === undefined) {
+      this.timer = setInterval(() => {
+        /* do some periodic work */
+      }, 1000);
+    }
+  }
+```
+
+{% endswitchable-sample %}
+
+
+We will start the timer in the `update` lifecycle callback.  The default
+implementation of `update` simply calls `render`, but since `update` is [_not_
+called during
+SSR](https://lit.dev/docs/templates/custom-directives/#differences-between-update()-and-render()),
+`update` is the right spot to start/subscribe to asynchronous tasks we don't
+want running on the server.
+
+Add the `update` callback. Call `ensureTimerStarted()` and then return the
+result of `render`. Note that we only want the timer running if the directive is
+currently connected (in some cases update could possibly be called when the
+directive has been disconnected, so you should always check `isConnected` before
```

Comment:
Is it guaranteed that `reconnected` will be called in this case? (Maybe add an `aside` at some point explaining that.)

Reply:
No, this was the months-long design quandary we had on this, and this is where we landed. A LitElement can be disconnected and still trigger a render, causing disconnected directives to be rendered. That does not cause directive reconnection, because nothing would subsequently disconnect them. You have to check `isConnected` in `update` before doing work that might require cleanup.

I moved this out to an aside.

Reply:
Can we file an issue for this? It feels like something we should revisit as it seems too hard to hold correctly. This seems like something we could easily address with another class `ConnectedDirective` that just vends `connected`?

Reply:
Note, this is fairly clearly documented: https://lit.dev/docs/templates/custom-directives/#async-directives